### PR TITLE
Update README dependency graph to match current setup.cfg

### LINKED ISSUE #1907: Update README dependency graph to match current setup.cfg
## Reason

The dependency graph at `img/dependencies.dot` (rendered to `dependencies.svg` and `dependencies.png`, shown in the README) is out of date. It still shows datashader, holoviews, rasterio, GDAL, GEOS, fiona, rtree, libspatialindex, and a shapely-via-fiona chain. None of those are runtime dependencies anymore:

- `xrspatial/` does not import datashader, holoviews, or rioxarray in the main package. datashader only appears in the `examples` extra.
- The README advertises "Free of GDAL / GEOS Dependencies" and ships its own GeoTIFF/COG reader and writer.
- The required dep set per `setup.cfg` is `numba, scipy, xarray, numpy, matplotlib, zstandard`.
- New optional runtime deps are missing from the chart: **kvikio** (GPUDirect Storage tile reads) and **nvCOMP** (GPU decompression), both lazily loaded in `xrspatial/geotiff/_gpu_decode.py`.

## Proposal

Rewrite `img/dependencies.dot` and re-render `dependencies.svg` and `dependencies.png` so the README image reflects what the project actually depends on.

**Required (green, solid edges)** — six entries from `install_requires`:
- numpy, xarray, numba, scipy, matplotlib, zstandard

**Optional (yellow, dashed edges)** — from `[options.extras_require]` and lazy imports:
- GPU compute: cupy, cuspatial, rtxpy
- GPU I/O acceleration: kvikio, nvcomp
- Distributed: dask, dask-geopandas
- Vector geometry: shapely, geopandas, awkward, spatialpandas
- Reprojection: pyproj
- GeoTIFF compression: deflate

**Removed from the chart:**
- datashader, holoviews (only in `examples` extra)
- rasterio, rioxarray, GDAL (replaced by built-in GeoTIFF reader/writer)
- GEOS, fiona, rtree, libspatialindex (no longer in the stack)
- pandas (transitive only; not used directly by xrspatial)

Flat layout — no Graphviz clusters, dashed edges from `xarray_spatial` to each optional, matching the style of the current diagram.

## Value

The chart in the README is one of the first things a new contributor or user reads. Showing nodes that are not actually in the dep tree is misleading and contradicts the "Free of GDAL / GEOS" claim a few lines below it.

## Files to Touch

- `img/dependencies.dot`
- `img/dependencies.svg`
- `img/dependencies.png`

### PR DESCRIPTION
**Pull request by brendancol**
## Summary

Rewrite `img/dependencies.dot` and re-render the SVG and PNG so the README diagram reflects what `xarray-spatial` actually depends on today.

- **Required (green, solid edges):** `numpy`, `xarray`, `numba`, `scipy`, `matplotlib`, `zstandard` — every entry in `install_requires`.
- **Optional (yellow, dashed edges):** `cupy`, `cuspatial`, `rtxpy`, `kvikio`, `nvcomp`, `dask`, `dask-geopandas`, `shapely`, `geopandas`, `awkward`, `spatialpandas`, `pyproj`, `deflate`.
- **Removed:** `datashader`, `holoviews`, `rasterio`, `GDAL`, `GEOS`, `fiona`, `rtree`, `libspatialindex`, `pandas`. The first four contradict the README's "Free of GDAL / GEOS Dependencies" line and the built-in GeoTIFF reader/writer; the rest are no longer in the tree.

Closes #1907.

## Test plan

- [x] `grep -E "datashader|holoviews|rasterio|gdal|geos|rtree|libspatialindex|fiona" img/dependencies.dot` returns no matches.
- [x] Each of `kvikio`, `nvcomp`, `cuspatial`, `rtxpy`, `matplotlib`, `zstandard`, `pyproj`, `deflate`, `awkward`, `dask_geopandas` appears in the .dot file.
- [x] `dot -Tsvg` and `dot -Tpng` regenerate the rendered images cleanly.
- [ ] Reviewer opens `img/dependencies.png` in the PR and confirms the layout reads well at README width.

### CODE REVIEW

**Comment by brendancol** [comment]
## PR Review: Update README dependency graph to match current setup.cfg

### Blockers (must fix before merge)
- None

### Suggestions (should fix, not blocking)
- [ ] `img/dependencies.dot:50,62,97` — `libnvcomp` and `kvikio` are not in `setup.cfg` under any extra. They are runtime dependencies of the GeoTIFF GPU path but installing the `gpu` extra (`cupy`, `cuspatial`) will not pull them in. Either add them to a `setup.cfg` extra (so the graph reflects what `pip install xarray-spatial[gpu]` actually installs), or add a short note in the README explaining the GPU GeoTIFF stack needs these installed separately. As is, the graph promises more than `setup.cfg` delivers.
- [ ] `img/dependencies.dot:28` — the second subgraph header was `subgraph required` before and is now `subgraph optional`. Graphviz does not enforce subgraph name uniqueness, but consider `cluster_required` / `cluster_optional` if you ever want the legend grouping to render as a visible cluster box. Pure cosmetic.

### Nits (optional improvements)
- [ ] `img/dependencies.dot:108` — `deflate` node label reads "(GeoTIFF Codec)". The PyPI package is `deflate` (libdeflate bindings); labelling it as a codec is fine but a tooltip/href to https://github.com/dcwatson/deflate would match the treatment of the other optional nodes that have `href=`.
- [ ] `img/dependencies.dot` — `datashader` is still listed in `setup.cfg` under the `examples` extra. Dropping it from the graph is defensible (examples are not runtime), but consider a one-line README caption clarifying that the graph shows runtime deps only, so readers do not infer that `examples` is gone.
- [ ] `img/dependencies.dot:51` — `dask` in `setup.cfg` is `dask[array]` (plus `dask[dataframe]` in `doc`). The graph node is just `dask`; fine as a simplification.

### What looks good
- Every `install_requires` entry (numpy, xarray, numba, scipy, matplotlib, zstandard) is represented as a green required node with an edge from `xarray_spatial`. Internal edges (`xarray -> numpy`, `numba -> numpy`, `scipy -> numpy`, `matplotlib -> numpy`) are accurate.
- Optional extras `optional` (awkward, geopandas, shapely, spatialpandas, rtxpy), `reproject` (pyproj), `geotiff` (deflate), `dask` (dask, dask-geopandas), and `gpu` (cupy, cuspatial) all show up as yellow dashed nodes.
- Removing `datashader`, `holoviews`, `rasterio`, `gdal`, `geos`, `fiona`, `rtree`, `libspatialindex`, `pandas` is correct — none of those are in the current tree, and the first four contradict the README's GDAL/GEOS-free claim and the built-in GeoTIFF reader.
- Color swap aquamarine -> palegreen for required nodes matches the legend wording.
- Renaming `nvcomp` to `libnvcomp` with the `libnvcomp.so` label avoids the misread that `pip install nvcomp` is the install path.
- Graphviz 9.0.0 regenerated the SVG cleanly; the `.dot` syntax parses.

### Checklist
- [ ] Algorithm matches reference/paper — N/A (docs-only)
- [ ] All implemented backends produce consistent results — N/A
- [ ] NaN handling is correct — N/A
- [ ] Edge cases are covered by tests — N/A
- [ ] Dask chunk boundaries handled correctly — N/A
- [ ] No premature materialization or unnecessary copies — N/A
- [ ] Benchmark exists or is not needed — N/A (not needed)
- [x] README feature matrix updated (if applicable) — dependency graph is the README artifact and it is updated
- [ ] Docstrings present and accurate — N/A

File: img/dependencies.dot [hunk 1]
```diff
@@ -6,58 +6,64 @@ edge [fontname = "helvetica"];
...
 subgraph required {
-	xarray_spatial -> datashader;
-	xarray_spatial -> numba;
+	xarray_spatial -> numpy;
 	xarray_spatial -> xarray;
-	xarray_spatial -> cupy;
+	xarray_spatial -> numba;
+	xarray_spatial -> scipy;
+	xarray_spatial -> matplotlib;
+	xarray_spatial -> zstandard;
```
[the remark this answers is not in the document]
Comment (brendancol) [reply img/dependencies.dot]: Addressed in 75e7379 (and the earlier 20e5677). Required nodes use `palegreen` fill and required edges now use `color=darkgreen` so the rendered diagram matches the "green, solid edges" legend in the PR description. Optional nodes stay yellow with dashed edges.

File: img/dependencies.dot [hunk 2]
```diff
@@ -6,58 +6,64 @@ edge [fontname = "helvetica"];
...
+ matplotlib [label=<<b>Matplotlib</b><br />(Plotting)> fillcolor=aquamarine style=filled];
+ zstandard [label=<<b>Zstandard</b><br />(Compression Codec)> fillcolor=aquamarine style=filled];
+
+ cupy [label=<<b>CuPy</b><br />(GPU-Ndarray)> fillcolor=lemonchiffon style=filled href="https://github.com/cupy/cupy"];
+ cuspatial [label=<<b>cuSpatial</b><br />(GPU Geometry)> fillcolor=lemonchiffon style=filled href="https://github.com/rapidsai/cuspatial"];
+ rtxpy [label=<<b>rtxpy</b><br />(GPU Ray Tracing)> fillcolor=lemonchiffon style=filled href="https://github.com/makepath/rtxpy"];
+ kvikio [label=<<b>KvikIO</b><br />(GPUDirect Storage I/O)> fillcolor=lemonchiffon style=filled href="https://github.com/rapidsai/kvikio"];
+ nvcomp [label=<<b>nvCOMP</b><br />(GPU Decompression)> fillcolor=lemonchiffon style=filled href="https://developer.nvidia.com/nvcomp"];
+ dask [label=<<b>Dask</b><br />(Distributed-Ndarray)> fillcolor=lemonchiffon style=filled href="https://dask.org"];
+ dask_geopandas [label=<<b>dask-geopandas</b><br />(Distributed GeoDataFrame)> fillcolor=lemonchiffon style=filled href="https://github.com/geopandas/dask-geopandas"];
```
[the remark this answers is not in the document]
Comment (brendancol) [reply img/dependencies.dot:62]: Addressed in 20e5677 and carried through 75e7379. The node is renamed from `nvcomp` to `libnvcomp`, its label is `libnvcomp.so`, and the `xarray_spatial -> libnvcomp` / `libnvcomp -> cupy` edges were updated to match. The href still points at the nvCOMP product page.

**Comment by brendancol** [comment]
## PR Review: Update README dependency graph to match current setup.cfg (re-review after rebase)

Rebased onto `origin/main` at `75e7379`. Both Copilot inline comments resolved in prior commits (`20e5677`, `211942c`) plus a small follow-up here.

### Scope
- `img/dependencies.dot`: explicit `edge [color=darkgreen]` inside `cluster_required` so required edges render in green, matching the legend wording in the PR description.
- Regenerated `img/dependencies.svg`, `img/dependencies.png`, and `docs/source/_static/img/dependencies.svg` from the updated `.dot`.

### Blockers
- None

### Suggestions
- None

### Verification
- `grep -E "datashader|holoviews|rasterio|gdal|geos|rtree|libspatialindex|fiona" img/dependencies.dot` returns no matches.
- `dot -Tsvg` and `dot -Tpng` produce the rendered images cleanly (graphviz 9.0.0).
- `libnvcomp` node + edges replace the prior `nvcomp` spelling; href still points at the nvCOMP product page.
- Required nodes use `palegreen` fill, optional nodes use `lemonchiffon`, optional edges keep `style=dashed`.

**Comment by brendancol** [comment]
Rebased onto current main at `6cd038e` to pick up the Python 3.14 CI matrix limit (`b1579f8`).
